test/rulesets/Base/array.jl

```diff
@@ -187,3 +187,68 @@ end
     test_rrule(fill, 55 + 0.5im, 5)
     test_rrule(fill, 3.3, (3, 3, 3))
 end
+
+@testset "findmin & findmax" begin
+    # Forward
+    test_frule(findmin, rand(10))
+    test_frule(findmax, rand(10))
+    @test @inferred(frule((nothing, rand(3,4)), findmin, rand(3,4))) isa Tuple{Tuple{Float64, CartesianIndex}, Tangent}
+    @test @inferred(frule((nothing, rand(3,4)), findmin, rand(3,4), dims=1)) isa Tuple{Tuple{Matrix, Matrix}, Tangent}
+    @test_skip test_frule(findmin, rand(3,4)) # StackOverflowError, CartesianIndex{2}(index::Tuple{Float64, Float64}) (repeats 79984 times) & TypeError: in new, expected Tuple{Int64, Int64}, got a value of type Tuple{Float64, Float64}
+    @test_skip test_frule(findmin, rand(3,4), output_tangent = (rand(), NoTangent()))
+    @test_skip test_frule(findmin, rand(3,4), fkwargs=(dims=1,))
+
+    # Reverse
+    test_rrule(findmin, rand(10), output_tangent = (rand(), false))
+    test_rrule(findmax, rand(10), output_tangent = (rand(), false))
+    @test [0 0; 0 5] == @inferred unthunk(rrule(findmax, [1 2; 3 4])[2]((5.0, nothing))[2])
+    @test_skip test_rrule(findmin, rand(10), output_tangent = (rand(), NoTangent()), check_inferred=false)  # DimensionMismatch from FiniteDifferences
+    @test_skip test_rrule(findmax, rand(5,3), output_tangent = (rand(), false), check_inferred=false)  # DimensionMismatch from FiniteDifferences
```

Comment:
Was still a bit unhappy about these tests.

Reply:
I think this would solve it: https://github.com/JuliaDiff/FiniteDifferences.jl/issues/188

Reply:
Any thoughts on this? Maybe should merge without these tests.

Reply:
I would suggest keeping as it is and adding a comment that a lot of dimension mismatches would be solved by fixing the JuliaDiff/FiniteDifferences.jl#188

Reply:
FWIW the current error here is:
```
julia> test_frule(findmin, rand(3,4))
test_frule: findmin on Matrix{Float64}: Error During Test at /Users/me/.julia/packages/ChainRulesTestUtils/73Y9Q/src/testers.jl:118
  Got exception outside of a @test
  iteration is deliberately unsupported for CartesianIndex. Use `I` rather than `I...`, or use `Tuple(I)...`
  Stacktrace:
    [1] error(s::String)
      @ Base ./error.jl:33
    [2] iterate(#unused#::CartesianIndex{2})
      @ Base.IteratorsMD ./multidimensional.jl:167
    [3] copyto!(dest::Vector{Int64}, src::CartesianIndex{2})
      @ Base ./abstractarray.jl:901
    [4] _collect(cont::UnitRange{Int64}, itr::CartesianIndex{2}, #unused#::Base.HasEltype, isz::Base.HasLength)
      @ Base ./array.jl:715
    [5] collect(itr::CartesianIndex{2})
      @ Base ./array.jl:709
    [6] test_approx(actual::CartesianIndex{2}, expected::CartesianIndex{2}, msg::Any; kwargs::Base.Pairs{Symbol, V, Tuple{Vararg{Symbol, N}}, NamedTuple{names, T}} where {V, N, names, T<:Tuple{Vararg{Any, N}}})
      @ ChainRulesTestUtils ~/.julia/packages/ChainRulesTestUtils/73Y9Q/src/check_result.jl:141
```